required/latex-lab/documentmetadata-support.dtx

```diff
@@ -0,0 +1,376 @@
+%         the core management code. By default the value is true.
+%
+%       \item[\texttt{firstaidoff}] This accepts a comma lists of keysword and disable the patches
+%       related to them. More information can be found in the documentation of
+%       \pkg{pdfmanagement-firstaid}.
+%
+%       \item[\texttt{testphase}] This key is used to load testphase code. The values it accepts
+%       and their effect will change over time, when testphase packages are added or
+%       removed when the code is moved into the kernel. Currently the accepted values are
+%       \texttt{tagpdf}, this load the tagpdf package, \texttt{headings}, this loads
+%       code which reimplements heading commands, and \texttt{ptagging} this loads code
+%       to allow paragraph tagging to work with engine other than luatex.
+%       \item[\texttt{activate}] This key is used to enable some document wide functions. It is
+%       currently in an experimental state. The values and their behaviour are subject to change.
+%       Currently the only value is |tagging|,
+%       which will do |\tagpdfsetup{activate,paratagging,interwordspace}|. It requires that
+%       \pkg{tagpdf} has been loaded first with the |testphase| key.
+%       \item[\texttt{debug}] This key activates some debug options. Currently only the
+%       keys |para| (with the default and only value |show|),
+%       and |log| (with the values of \pkg{tagpdf}) and |uncompress| (which does the same
+%       as |uncompress| as main key)  are known.
+%    \end{description}
+%
+%
+%
+% \StopEventually{\setlength\IndexMin{200pt}  \PrintIndex  }
+%
+%
+% \section{The Implementation}
+%
+%    \begin{macrocode}
+%<*code>
+%    \end{macrocode}
+%
+%    \begin{macrocode}
+
+\RequirePackage{pdfmanagement-testphase}
+%    \end{macrocode}
+%
+%    \begin{macrocode}
+\ExplSyntaxOn\makeatletter
+%    \end{macrocode}
+
+
+
+%  \begin{macro}{\DocumentMetadata}
+%    
+%    \cs{DocumentMetadata} should not be used after
+%    \cs{documentclass} so we error in this case.
+%    It can be used more than once
+%     but follow-up calls should not do the initialization code.
+%    \begin{macrocode}
+\cs_set_protected:Npn \DocumentMetadata #1
+  {
+    \cs_if_eq:NNTF \documentclass \@twoclasseserror
+      { \msg_error:nn { meta } { after-class } }
+      {
+%    \end{macrocode}
+%
+%    The wanted backend must be detected first, we read the init
+%    key and then force the loading by either loading l3pdf (+ expl3)
+%    or only expl3.  The backend contains now management command, so
+%    the boolean should true first.
+%    \begin{macrocode}
+        \keys_set_groups:nnn { document / metadata} {init}{ #1 }
+        %should be loaded after the backend is set, and only if not in the kernel
+        \bool_gset_true:N \g__pdfmanagement_active_bool
+        \cs_if_free:NTF \pdf_uncompress:
+           {\RequirePackage{l3pdf}}
+%FMi why load expl3? is there a better way to force the backend loading?          
```

Comment:
` \sys_load_backend:n { <backend> }` should work as long as we know what `<backend>` is at this point.

Reply:
`\sys_load_backend:n{}` will go with the default one.

Reply:
I will remove the test and loading of `l3pdf` from ltdocinit, imho it is now long enough in the kernel.  

Instead of loading expl3 one could imho do now something like this
~~~~
\sys_load_backend:n { <value of DocumentMetadata backend key>}
\str_if_exist:NF \c_sys_backend_str
  { \sys_load_backend:n { } }
~~~~

Currently the backend key simply pass its value to the backend key of expl3, and so is automatically in sync and accept the same values. So if we change this one will have to adapt the value in both if needed.

Reply:
well, could you please to the change (now or at a later stage)? This code is only ever used if the format contains ltmeta which means it does have expl3 inside so we should never do the package loading (which gives option clashes if loaded later again to set options).

Reply:
I pushed the change, I will have to adapt some of the tests, they fail as a luatex font number has changed, but this is unrelated.

Reply:
@u-fischer I had hoped you push the change here :-( ... ok reverse engineer time then